pkg/controllers/machinesetsync/machineset_sync_controller.go

```diff
@@ -98,88 +105,218 @@ func (r *MachineSetSyncReconciler) SetupWithManager(mgr ctrl.Manager) error {
 
-		mapiMachineSetNotFound = true
+		mapiMachineSet = nil
 	} else if err != nil {
-		logger.Error(err, "Failed to get MAPI MachineSet")
-		return ctrl.Result{}, fmt.Errorf("failed to get MAPI MachineSet: %w", err)
+		logger.Error(err, "Failed to get MAPI machine set", "name", name)
+		return nil, nil, fmt.Errorf("failed to get MAPI machine set: %w", err)
 	}
 
-	// Get the corresponding CAPI Machine.
+	// Fetch CAPI MachineSet
 	capiMachineSet := &capiv1beta1.MachineSet{}
-	capiNamespacedName := client.ObjectKey{
-		Namespace: r.CAPINamespace,
-		Name:      req.Name,
-	}
 
-	if err := r.Get(ctx, capiNamespacedName, capiMachineSet); apierrors.IsNotFound(err) {
-		logger.Info("CAPI MachineSet not found")
+	if err := r.Get(ctx, client.ObjectKey{Namespace: r.CAPINamespace, Name: name}, capiMachineSet); apierrors.IsNotFound(err) {
+		logger.Info("CAPI machine set not found", "name", name)
 
-		capiMachineSetNotFound = true
+		capiMachineSet = nil
 	} else if err != nil {
-		logger.Error(err, "Failed to get CAPI MachineSet")
-		return ctrl.Result{}, fmt.Errorf("failed to get CAPI MachineSet: %w", err)
+		logger.Error(err, "Failed to get CAPI machine set", "name", name)
+		return nil, nil, fmt.Errorf("failed to get CAPI machine set: %w", err)
 	}
 
-	if mapiMachineSetNotFound && capiMachineSetNotFound {
-		logger.Info("CAPI and MAPI MachineSets not found, nothing to do")
-		return ctrl.Result{}, nil
-	}
+	return mapiMachineSet, capiMachineSet, nil
+}
 
-	// If the MachineSet only exists in CAPI, we don't need to sync back to MAPI.
-	if mapiMachineSetNotFound {
-		logger.Info("Only CAPI MachineSet found, nothing to do")
-		return ctrl.Result{}, nil
-	}
+// syncMachineSets synchronizes MachineSets based on the authoritative API.
+func (r *MachineSetSyncReconciler) syncMachineSets(ctx context.Context, mapiMachineSet *machinev1beta1.MachineSet, capiMachineSet *capiv1beta1.MachineSet) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
 
 	switch mapiMachineSet.Status.AuthoritativeAPI {
 	case machinev1beta1.MachineAuthorityMachineAPI:
-		return r.reconcileMAPIMachineSettoCAPIMachineSet(ctx, mapiMachineSet, capiMachineSet)
+		return r.reconcileMAPIMachineSetToCAPIMachineSet(ctx, mapiMachineSet, capiMachineSet)
 	case machinev1beta1.MachineAuthorityClusterAPI:
-		return r.reconcileCAPIMachineSettoMAPIMachineSet(ctx, capiMachineSet, mapiMachineSet)
+		return r.reconcileCAPIMachineSetToMAPIMachineSet(ctx, capiMachineSet, mapiMachineSet)
 	case machinev1beta1.MachineAuthorityMigrating:
-		logger.Info("machine currently migrating", "machine", mapiMachineSet.GetName())
+		logger.Info("machine set is currently migrating", "machine set", mapiMachineSet.GetName())
 		return ctrl.Result{}, nil
 	default:
-		logger.Info("machine AuthoritativeAPI has unexpected value", "AuthoritativeAPI", mapiMachineSet.Status.AuthoritativeAPI)
+		logger.Info("Unexpected value for AuthoritativeAPI", "AuthoritativeAPI", mapiMachineSet.Status.AuthoritativeAPI)
```

Comment:
Is name/namespace of the resource inherited and reported in the k/v for this "unexpected value" log line?

Reply:
Yup :)